modules/gapi/src/backends/render/grenderocvbackend.cpp

```diff
@@ -0,0 +1,141 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+#include "precomp.hpp"
+
+#include <functional>
+#include <unordered_set>
+
+#include <ade/util/algorithm.hpp>
+
+#include <ade/util/range.hpp>
+#include <ade/util/zip_range.hpp>
+#include <ade/util/chain_range.hpp>
+#include <ade/typed_graph.hpp>
+
+#include <opencv2/gapi/gcommon.hpp>
+#include <opencv2/gapi/garray.hpp>
+#include <opencv2/gapi/util/any.hpp>
+#include <opencv2/gapi/gtype_traits.hpp>
+
+#include "compiler/gobjref.hpp"
+#include "compiler/gmodel.hpp"
+
+#include "api/gbackend_priv.hpp" // FIXME: Make it part of Backend SDK!
+#include "api/render_ocv.hpp"
+
+#include "backends/render/grenderocvbackend.hpp"
+
+
+using GRenderModel = ade::TypedGraph
+    < cv::gimpl::render::ocv::RenderUnit
+    >;
+
+// FIXME: Same issue with Typed and ConstTyped
+using GConstRenderModel = ade::ConstTypedGraph
+    < cv::gimpl::render::ocv::RenderUnit
+    >;
+
+cv::gimpl::render::ocv::GRenderExecutable::GRenderExecutable(const ade::Graph &g,
+                                                        const std::vector<ade::NodeHandle> &nodes)
+    : m_g(g), m_gm(m_g) {
+        GConstRenderModel gcm(m_g);
+        for (auto &nh : nodes)
+        {
+            switch (m_gm.metadata(nh).get<NodeType>().t)
+            {
+                case NodeType::OP:
+                    if (this_nh == nullptr) {
+                        this_nh = nh;
+                    }
+                    else
+                        util::throw_error(std::logic_error("Multi-node rendering is not supported!"));
+                    break;
+                default: util::throw_error(std::logic_error("Unsupported NodeType type"));
+            }
+        }                  
```

Comment:
I'd rewrite it with two `find_if`s. The first one would be used to obtain `this_nh` and the second one to check if that one was the only one.
Yes, IE backend written like here but it also handles DATA things

Reply:
Fixed